lib/_http_client.js

```diff
@@ -197,7 +201,27 @@ function ClientRequest(input, options, cb) {
   }
   this.insecureHTTPParser = insecureHTTPParser;
 
-  this.path = options.path || '/';
+  path = options.path;
+  if (path) {
+    if (method === 'CONNECT') {
+      if (path.charAt(0) === '/') {
```

Comment:
```suggestion
      if (path.startsWith('/')) {
```
Arguably a little more idiomatic for core code.